draftjs_exporter/composite_decorators.py

```diff
@@ -0,0 +1,60 @@
+from __future__ import absolute_import, unicode_literals
+
+import cgi
+import re
+
+
+class CompositeDecorator():
+
+    # Override this to describe the pattern to be replaced
+    SEARCH_RE = re.compile('$^')
+
+    # Implemente this to generate a html string to replace the matched text
+    def replace(self, match):
+        raise NotImplementedError
+
+    def process(self, text, parent=None):
+        self.parent = parent
+        result = ""
+        begin = final = 0
+        for i in self.SEARCH_RE.finditer(text):
+            begin, end = i.span()
+            # append other content
+            result += text[final:begin]
+            # append replaced content
+            result += self.replace(i)
```

Comment:
I think this is just the right implementation except for one thing – it treats HTML as a string instead of using the power of `DOM` / BeautifulSoup to create proper DOM nodes. Is there a way to change this so it uses nodes instead of string concat?

My naive attempt would be something like:

```python
result = DOM.create_document_fragment()

[...] 

DOM.append_child(result, DOM.create_text_node(text[final:begin]))
DOM.append_child(result, self.replace(i))

[...]

DOM.append_child(result, DOM.create_text_node(text[final:]))
```